sdk/documentintelligence/Azure.AI.DocumentIntelligence/src/TrainingOperation.cs

```diff
@@ -46,38 +47,23 @@ public override async ValueTask<Response> UpdateStatusAsync(CancellationToken ca
 
         private string GetOperationId()
         {
-            const string OperationIdNotFoundErrorMessage = "The operation ID was not present in the service response.";
             var response = GetRawResponse();
 
-            // If the "operation-location" header is present, it means the latest response came from a POST request,
-            // so the operation ID can be extracted from the header.
-            // Otherwise, the latest response came from a GET request, so the operation ID can be obtained directly
-            // from the response's content.
-
             if (response.Headers.TryGetValue("operation-location", out string operationLocation))
             {
-                // "operation-location" header pattern:
-                // https://<endpoint>/documentintelligence/operations/<operationId>?api-version=<version>
-                var match = Regex.Match(operationLocation, @"[^:]+://[^/]+/documentintelligence/operations/([^?]+)");
+                // Location header pattern:
+                // https://<endpoint>/documentintelligence/<api>/<operationId>?api-version=<version>
+                var match = Regex.Match(operationLocation, @"[^:]+://[^/]+/documentintelligence/.+/([^?/]+)");
```

Comment:
Do we need this much rigor?  Couldn't we do something more lightweight like:

```csharp
var location = operationLocation.AsSpan();
var lastSlash = location.LastIndexOf('/');
var lastQuery = location.LastIndexOf('?');

// Not in the correct format.
if (lastSlash < 0 || lastQuery < 0 || lastQuery <= lastSlash)
{
    return null;
}

return location.Slice(lastSlash + 1, lastQuery - lastSlash - 1).ToString();
```

Reply:
I think that with regex we can make it a bit safer. If for any reason the service changes something and starts putting a slash before the start of the query:
```
https://<endpoint>/documentintelligence/<api>/<operationId>/?api-version=<version>
```
Or in case they don't return the query at all:
```
https://<endpoint>/documentintelligence/<api>/<operationId>
```
These scenarios are easily caught by the regex, but we would need to need to be smarter if we wanted to parse the operation ID using indexes.

Reply:
Fair.   If we don't have a strong contract on the format of the URI, the added weight of the RegEx is justified.